bootstrap.sh

```diff
@@ -1,39 +1,505 @@
-        sudo apt-get update -qq && sudo apt-get install -qqy google-cloud-sdk
+        echoSuccess "gcloud installed"
+        gcloud --version
+        echo ""
+        return 0
     fi
-else
-    echo "gcloud already installed"
-    gcloud --version    
+}
+
+check_gcp() {
+    { # Try
+        if [[ -z $(command -v gcloud) ]]; then
+            echoInfo "====> Installing gcloud sdk..."
+            if [[ "$OSTYPE" == "linux-gnu" ]]; then
+                install_gcp "linux"
+            elif [[ "$OSTYPE" == "darwin"* ]]; then
+                # Mac OSX
+                install_gcp "darwin"
+            elif [[ "$OSTYPE" == "cygwin" ]]; then
+                # POSIX compatibility layer and Linux environment emulation for Windows
+                install_gcp "windows"
+            elif [[ "$OSTYPE" == "msys" ]]; then
+                # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+                install_gcp "windows"
+            elif [[ "$OSTYPE" == "win32" ]]; then
+                # I'm not sure this can happen.
+                install_gcp "windows"
+            elif [[ "$OSTYPE" == "freebsd"* ]]; then
+                install_gcp "linux"
+            else
+                help_install_gcp_sdk
+                return 1
+            fi
+        else
+            echoSuccess "gcloud found in path!"
+            gcloud --version    
+            echo ""
+            return 0
+        fi
+        return $?
+    } || { # Catch
+        help_install_gcp_sdk
+        return 1
+    }
+}
+
+help_install_tf() {
+    echoError "We could not automatically install terraform"
+    echoInfo "Please download the relevant zip package according to your operating system from here: ${TERRAFORM_PACKAGE_URL}"
+    echoInfo "Terraform is distributed as a single binary. Install Terraform by unzipping it and moving it to a directory included in your system's PATH ."
+}
+
+install_tf() {
+    echoInfo "====> Installing terraform..."
+    if [[ "$1" == "windows" ]]; then
+        help_install_tf
+        return 1
+    fi
+    if [[ -z "$(command -v unzip)" ]]; then
+        {
+            sudo apt-get update -qy && sudo apt-get install -qy unzip > /dev/null
+        } || {
+            sudo apt update -qy && sudo apt install -qy unzip > /dev/null
+        } || {
+            sudo yum update -qy && sudo yum install -qy unzip > /dev/null
+        } || {
+            brew tap homebrew/dupes && brew install unzip > /dev/null
+        } || {
+            echoError "Could not install unzip"
+            help_install_tf
+            return 1
+        }
+        
+    fi
+    curl -fSL -o terraform.zip https://releases.hashicorp.com/terraform/"$TERRAFORM_VERSION"/terraform_"$TERRAFORM_VERSION"_"$1"_amd64.zip
+    sudo mkdir -p "$LIB_LOCATION"/
+    sudo unzip -q terraform.zip -d "$LIB_LOCATION"/terraform
+    rm -f terraform.zip
+    sudo ln -Ffs "$LIB_LOCATION"/terraform/terraform /usr/local/bin/terraform
```

Comment:
isn't terraform just a single executable file? should be enough to slap it in /usr/local/bin

Reply:
Imho, using a symbolic link is the standard way of managing adding binaries to the path.